iothub/device/src/Twin/PropertyCollection.cs

```diff
@@ -0,0 +1,103 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public abstract class PropertyCollection : IEnumerable<KeyValuePair<string, object>>
+    {
+        private const string VersionName = "$version";
+
+        private protected PropertyCollection()
```

Comment:
why private?

Reply:
this constructor is no longer required. I've reworked this and added in a null check for _properties